### PR TITLE
[ONNX] add jit pass for lists

### PR DESCRIPTION
**Pull request by KsenijaS**
Add jit preprocessing pass for adding int lists.

### CODE REVIEW

File: torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp [hunk 1]
```diff
@@ -104,10 +104,40 @@ static void FuseWithListUnpack(Block* b) {
...
+      auto in1 = it->input(0)->node();
+      auto in2 = it->input(1)->node();
+      if (!in1->output()->type()->cast<ListType>() ||
+          !in2->output()->type()->cast<ListType>()) {
+        continue;
+      }
+
+      TypePtr elem = in2->output()->type()->cast<ListType>()->getElementType();
+      if (elem->cast<IntType>()) {
+        std::cout << in1->kind() << " " << in2->kind();
```
Comment (neginraoof) [inline torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp:122]: Remove debug cout.

**Comment by neginraoof** [comment]
@KsenijaS 
Test failures are related. Please take a look.

File: torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp [hunk 2]
```diff
@@ -104,10 +104,37 @@ static void FuseWithListUnpack(Block* b) {
   }
 }
 
+static void ReplaceAddWithConcat(Block* b) {
```
Comment (neginraoof) [inline torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp:107]: Please add some comments on the top about the pass and the graph updates.

**Review by neginraoof — APPROVED** [review]
LGTM, thanks.
Can you please rebase again to merge changes that enable scripting tests, to see if these tests would pass for scripting?

**Comment by KsenijaS** [comment]
@bzinodev Can we merge this PR?

File: torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp [hunk 3]
```diff
@@ -104,10 +104,63 @@ static void FuseWithListUnpack(Block* b) {
...
+      ReplaceAddWithConcat(child_block);
+    }
+    if (it->kind() == aten::add) {
+      if (!it->input(0)->type()->cast<ListType>() ||
+          !it->input(1)->type()->cast<ListType>()) {
+        continue;
+      }
+
+      TypePtr elem = it->input(0)->type()->cast<ListType>()->getElementType();
+      if (elem->cast<IntType>()) {
```
[the remark this answers is not in the document]
Comment (KsenijaS) [reply torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp:145]: Tensor types are already covered by this PR: https://github.com/pytorch/pytorch/pull/41888
I think that condition that checks if list type is not Tensor is more accurate

**Review by BowenBao — APPROVED** [review]

**Review by bzinodev — APPROVED** [review]

**Comment by KsenijaS** [comment]
@bzinodev I rebased and resolved the merge conflict. Can we merge this PR? Thanks!

**Comment by KsenijaS** [comment]
@bzinodev  Some internal Facebook tests are failing, does this PR needs some changes or it can be merged as is. Thanks!

**Comment by KsenijaS** [comment]
Can we merge this please? Thanks!

**Pull request closed without merging**